x/margin/keeper/keeper.go

```diff
@@ -0,0 +1,104 @@
+package keeper
+
+import (
+	"github.com/Sifchain/sifnode/x/margin/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+type Keeper struct {
+	storeKey   sdk.StoreKey
+	cdc        codec.BinaryCodec
+	bankKeeper types.BankKeeper
+	clpKeeper  types.CLPKeeper
+	paramStore paramtypes.Subspace
+}
+
+func NewKeeper(storeKey sdk.StoreKey,
+	cdc codec.BinaryCodec,
+	bankKeeper types.BankKeeper,
+	clpKeeper types.CLPKeeper) KeeperI {
+
+	return Keeper{bankKeeper: bankKeeper, clpKeeper: clpKeeper, storeKey: storeKey, cdc: cdc}
+}
+
+func (k Keeper) SetMTP(ctx sdk.Context, mtp *types.MTP) error {
+	if err := mtp.Validate(); err != nil {
+		return err
+	}
+	store := ctx.KVStore(k.storeKey)
+	key := types.GetMTPKey(mtp.CollateralAsset, mtp.Address)
+	store.Set(key, k.cdc.MustMarshal(mtp))
+	return nil
+}
+
+func (k Keeper) GetMTP(ctx sdk.Context, symbol string, mtpAddress string) (types.MTP, error) {
+	var mtp types.MTP
+	key := types.GetMTPKey(symbol, mtpAddress)
+	store := ctx.KVStore(k.storeKey)
+	if !store.Has(key) {
+		return mtp, types.ErrMTPDoesNotExist
+	}
+	bz := store.Get(key)
+	k.cdc.MustUnmarshal(bz, &mtp)
+	return mtp, nil
+}
+
+func (k Keeper) GetMTPIterator(ctx sdk.Context) sdk.Iterator {
+	store := ctx.KVStore(k.storeKey)
+	return sdk.KVStorePrefixIterator(store, types.MTPPrefix)
+}
+
+func (k Keeper) GetMTPs(ctx sdk.Context) []*types.MTP {
+	var mtpList []*types.MTP
+	iterator := k.GetMTPIterator(ctx)
+	defer iterator.Close()
```

Comment:
Not sure if it can ever be triggered , but it still might be a good idea to handle the error returned , maybe a panic . 
The same for all the other places this is used